### PR TITLE
#166237761- Setup PostgreSQL Database 

### PR DESCRIPTION
**Pull request by chialuka**
#### What does this PR do?

- Setup PostgreSQL cloud-hosted database for the project

- Setup databases for testing, development, and production and add the URLs to the project

- Setup configuration for the DB instances

- Write migration scripts for the databases

#### Description of Task to be completed?

- Ensure databases for testing, development, and production are set up and running correctly

- Ensure migration scripts work correctly

#### How should this be manually tested?

- Clone the repo and run the relevant scripts described in the `package.json` file

- Navigate to the relevant database and confirm the success of the script run

#### Any background context you want to provide?
This PR creates PostgreSQL database instances for the project. It creates three databases and provides migration scripts also

#### What are the relevant pivotal tracker stories?
[#166237761](https://www.pivotaltracker.com/story/show/166237761)

### CODE REVIEW

File: server/config/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  production: {
+    url: DB_URL_PROD
+  }
+};
+
+const connectionString = dbUrl[NODE_ENV].url;
+
+const pool = new Pool({ connectionString });
+
+export default pool;
```
Comment (Nerocodes) [inline server/config/index.js:26]: Awesome!! This logic is good, instead of using an if or a switch statement this is preferable. Good job.

**Review by Nerocodes — APPROVED** [review]
Good job. I like the approach in your config index file.

File: server/config/index.js (hunk 1 above)
Comment (chialuka) [reply server/config/index.js:26]: Thank you Nero

**Review by ayodejiAA — COMMENTED** [review]
Good effort @LukasChiama . Like what you did with the connectionString variables. Really cool. 

Now how can we solve the hound  errors?

**Comment by chialuka** [comment]
@Xwebyna, I honestly don't know. The sourceType is given in the eslint configuration file as module. I don't know where the hound error is coming from

File: server/config/index.js (hunk 1 above)
Comment (tolulope-od) [reply server/config/index.js:26]: This is really cool @LukasChiama Some JavaScript ninja kung fu going on here. lol! I like this, it's a very elegant implementation!

File: index.js [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
+import routes from './server/routes';
+import swaggerDoc from './hello-books-swagger.json';
 
 dotenv.config();
 const port = process.env.PORT || 4000;
 const app = express();
 
 const { log } = console;
 
+log(process.env.NODE_ENV);
```
Comment (tolulope-od) [inline index.js]: Great stuff @LukasChiama very clean code as usual, however, is there any particular reason why you're logging the environment to the console or is this just as a heads up to be aware of the environment we're working in?

Comment (chialuka) [reply index.js]: O boy! I totally missed that. I used it to confirm the setup was working properly and then forgot to delete before pushing. Thanks for catching this. I'll take it away immediately.

**Review by obayomi96 — APPROVED** [review]
Great work here @LukasChiama .  This is clean.

**Review by abdulfatai360 — APPROVED** [review]
Great job @LukasChiama . I foresee a merge conflict in the `test script` with my PR. We'll cross the bridge when we get there..

Awesome job!

**Review by ayodejiAA — APPROVED** [review]

**Pull request merged**